### PR TITLE
Cpu cores support

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref: #174 

Adds ability to configure cpu cores in resource vcd_vapp_vm. E.g.

```
resource "vcd_vapp" "web" {
  name = "Web"
}
resource "vcd_vapp_vm" "web1" {
  name = "web-01-${count.index}"
  vapp_name = "${vcd_vapp.web.name}"
  catalog_name  = "testCatalog"
  template_name = "photon-hw11"

  network_name = "myOrgNet1"
  memory        = 1024
  cpus          = 4
  cpu_cores     = 2
  ip            = "allocated"
  count = 1

  depends_on = ["vcd_vapp.web"]
}
```

### CODE REVIEW

File: website/docs/r/vapp_vm.html.markdown [hunk 1]
```diff
@@ -77,8 +78,9 @@ The following arguments are supported:
 * `name` - (Required) A unique name for the vApp
```
Comment (Didainius) [inline website/docs/r/vapp_vm.html.markdown]:
```suggestion
* `name` - (Required) A unique name for the VM
```

File: vcd/resource_vcd_vapp_vm_test.go [hunk 2]
```diff
@@ -156,7 +156,8 @@ resource "vcd_vapp_vm" "{{.VmName}}" {
   catalog_name  = "{{.Catalog}}"
   template_name = "{{.CatalogItem}}"
   memory        = 1024
-  cpus          = 1
+  cpus          = 2
+  cpu_cores		= 1
```
Comment (Didainius) [inline vcd/resource_vcd_vapp_vm_test.go]: Not sure if it's github or the "= 1" part is misaligned

File: vcd/resource_vcd_vapp_vm.go [hunk 3]
```diff
@@ -441,7 +445,18 @@ func resourceVcdVAppVmUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		if d.HasChange("cpus") {
 			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-				task, err := vm.ChangeCPUCount(d.Get("cpus").(int))
+				var task govcd.Task
+				var err error
+				if d.HasChange("cpu_cores") {
```
Comment (Didainius) [inline vcd/resource_vcd_vapp_vm.go]: We won't get here if someone changes just 'cpu_cores' without altering 'cpus'

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +18,7 @@ FEATURES:
 IMPROVEMENTS:
 
 * resource/vcd_inserted_media now supports force ejecting on running VM [GH-184]
+* resource/vcd_vapp_vm now support cpu cores configuration [GH-168]
```
Comment (lvirbalas) [inline CHANGELOG.md]: Let's capitalise: cpu -> CPU

Comment (lvirbalas) [reply CHANGELOG.md]: Wrong GH-168 number? It shows up as "Update repository to use Go modules".

File: vcd/resource_vcd_vapp_vm.go [hunk 5]
```diff
@@ -441,7 +445,18 @@ func resourceVcdVAppVmUpdate(d *schema.ResourceData, meta interface{}) error {
...
+					if d.Get("cpu_cores") != nil {
+						coreCounts := d.Get("cpu_cores").(int)
+						task, err = vm.ChangeCPUCountWithCore(d.Get("cpus").(int), &coreCounts)
+					} else {
+						return resource.RetryableError(fmt.Errorf("missing cpu_core value: %#v", err))
+					}
+				} else {
+					task, err = vm.ChangeCPUCount(d.Get("cpus").(int))
+				}
```
Comment (lvirbalas) [inline vcd/resource_vcd_vapp_vm.go]: Please add a test that changes the CPU cores.

File: vcd/resource_vcd_vapp_vm.go (hunk 3 above)
Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm.go]: Fixed, though it worked somehow.

File: vcd/resource_vcd_vapp_vm_test.go (hunk 2 above)
Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm_test.go]: Github issue, as pasted after `terraform fmt`

File: website/docs/r/vapp_vm.html.markdown (hunk 1 above)
Comment (vbauzysvmware) [reply website/docs/r/vapp_vm.html.markdown]: Fixed

File: CHANGELOG.md (hunk 4 above)
Comment (vbauzysvmware) [reply CHANGELOG.md]: Yes, good catch

File: vcd/resource_vcd_vapp_vm.go (hunk 5 above)
Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm.go]: resource_vcd_vapp_vm_test.go covers that

Comment (lvirbalas) [reply vcd/resource_vcd_vapp_vm.go]: For a moment I thought I was looking at master, so went and double checked the stash branch - the test only covers `cpu`, there's nothing around `cores`.

Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm.go]: in  vcd/resource_vcd_vapp_vm_test.go change:
```
  cpus          = 2
  cpu_cores		= 1
```

File: vcd/resource_vcd_vapp_vm_test.go (hunk 2 above)
Comment (dn1s) [reply vcd/resource_vcd_vapp_vm_test.go]: It's not a github issue. It is space vs tabs. Go fmt does not convert from one into the other which results into this.

Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm_test.go]: Looked in wrong place -fixed now.

File: website/docs/r/vapp_vm.html.markdown [hunk 6]
```diff
@@ -74,11 +75,12 @@ resource "vcd_vapp_vm" "web2" {
...
 * `vapp_name` - (Required) The vApp this VM should belong to.
-* `name` - (Required) A unique name for the vApp
+* `name` - (Required) A unique name for the VM
 * `catalog_name` - (Required) The catalog name in which to find the given vApp Template
 * `template_name` - (Required) The name of the vApp Template to use
-* `memory` - (Optional) The amount of RAM (in MB) to allocate to the vApp
-* `cpus` - (Optional) The number of virtual CPUs to allocate to the vApp
+* `memory` - (Optional) The amount of RAM (in MB) to allocate to the VM
+* `cpus` - (Optional) The number of virtual CPUs to allocate to the VM. Socket count is a result of: virtual logical processors/cores per socket
+* `cpu_cores` - (Optional) The number of cores per socket
```
Comment (lvirbalas) [inline website/docs/r/vapp_vm.html.markdown]: We need to add the version "tag" to this parameter as it's new only to this release, while the resource is not new. Please also think whether there may be other version tags on parameters that we may have missed on old resources.

Comment (vbauzysvmware) [reply website/docs/r/vapp_vm.html.markdown]: Added

**Pull request merged**